packages/sds/src/message_channel/events.ts

```diff
@@ -55,6 +59,132 @@ export class Message implements proto_sds_message.SdsMessage {
   }
 }
 
+export class SyncMessage extends Message {
+  public constructor(
+    public messageId: string,
+    public channelId: string,
+    public senderId: string,
+    public causalHistory: proto_sds_message.HistoryEntry[],
+    public lamportTimestamp: number,
+    public bloomFilter: Uint8Array<ArrayBufferLike> | undefined,
+    public content: undefined,
+    /**
+     * Not encoded, set after it is sent, used to include in follow-up messages
+     */
+    public retrievalHint?: Uint8Array | undefined
+  ) {
+    super(
+      messageId,
+      channelId,
+      senderId,
+      causalHistory,
+      lamportTimestamp,
+      bloomFilter,
+      content,
+      retrievalHint
+    );
+  }
+}
+
+export function isSyncMessage(
+  message: Message | ContentMessage | SyncMessage | EphemeralMessage
+): message is SyncMessage {
+  return Boolean(
+    "lamportTimestamp" in message &&
+      typeof message.lamportTimestamp === "number" &&
+      (message.content === undefined || message.content.length === 0)
+  );
+}
+
+export class EphemeralMessage extends Message {
```

Comment:
i think this class, `ContentMessage, and `SyncMessage` should be in their own file